### ISSUE TITLE
Training stuckafter step 'Pre-filling staging area ...'

### ISSUE THREAD

**Issue by MXCao** [ppwwyyxx/tensorpack#617]
Hi, I used your imagenet-resnet-keras.py to train on my own dataset. And I modified the ilsvrc.py to make my own dataflow. 
It can run the 'Pre-filling staging area ...' step, after that a model will be saved. But the program then stucks at next step. It does not throw any error. 
<img width="988" alt="screen shot 2018-01-31 at 11 19 20 am" src="https://user-images.githubusercontent.com/32104513/35603531-b92df5d8-0678-11e8-9f70-505f1784dafd.png">

I found there's an error during the model initialisation: 
`ERR [MultiThreadMapData] buffer_size cannot be larger than the size of the DataFlow!`
<img width="1015" alt="screen shot 2018-01-31 at 11 21 53 am" src="https://user-images.githubusercontent.com/32104513/35603595-00099e30-0679-11e8-84d0-700f6419b896.png">
This may be the reason, i tried to trace the error but couldn't figure out. Could you please give me some advices? Thanks!

**Comment by ppwwyyxx** [comment]
It is the reason. And it has already told you why it failed.
You can print the size of your dataflow by `print(dataflow.size())`.

**Comment by MXCao** [comment]
I fixed the problem because i noticed that the default buffer_size was set to be 2000. Howerver, my validation set has only 1500 images. That's why it failed.
